src/tiny-bang-interpreter/tiny_bang_interpreter.ml

```diff
@@ -107,9 +107,8 @@ let rec application_match env x_fn x_arg : clause list option =
   logger `debug ("Lookup: " ^ pretty_var x_fn ^ "\n");
   match lookup env x_fn with
   | Onion_value(_, x_left, x_right) ->
-    Option.map_default Option.some
-      (application_match env x_left x_arg)
-      (application_match env x_right x_arg)
+    BatEnum.peek @@
```

Comment:
Previously:
```
let f = (fun () -> 1) & (fun () -> 2) in 
f 10
;;
```
(which translated to)
```
__2 = __6  { __6 = () } -> __5 = 1;
__1 = __4  { __4 = () } -> __3 = 2;
f = __2 & __1;
__8 = f;
__7 = 10;
__0 = __8 __7
```
outputted 2, not 1. This fixes the problem.